r/tests/testthat/test-dataset.R

```diff
@@ -943,10 +943,42 @@ test_that("Dataset writing: from RecordBatch", {
   )
 })
 
+test_that("Writing a dataset: Ipc format options & compression", {
+  skip_on_os("windows") # https://issues.apache.org/jira/browse/ARROW-9651
+  ds <- open_dataset(csv_dir, partitioning = "part", format = "csv")
+  dst_dir <- make_temp_dir()
+
+  metadata <- c(hello = "world", eh = "!")
+  codec <- NULL
+  if (codec_is_available("zstd")) {
+    codec <- Codec$create("zstd")
+  }
+
+  write_dataset(ds, dst_dir, format = "feather", metadata = metadata, codec = codec)
+  expect_true(dir.exists(dst_dir))
+
+  file <- ds$filesystem$OpenInputStream(paste(dst_dir, dir(dst_dir)[[1]], sep = "/"))
+  expect_equivalent(metadata, RecordBatchFileReader$create(file)$metadata)
```

Comment:
```suggestion
  file <- dir(dst_dir, full.names = TRUE)[[1]]
  expect_equivalent(read_feather(file, as_data_frame = FALSE)$metadata, metadata)
```

Reply:
Unfortunately this doesn't work; the footer metadata is distinct from the schema metadata (which is what this change would check)

Reply:
What is footer metadata for? 

Would it make sense/be ok with you to defer wiring up footer metadata in R for this release? I don't understand it enough to advise on how to expose it in R in a way that is useful and not confusing.

Reply:
alright